### PR TITLE
[Refactor] Add CatalogAnalyzer

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

related with https://github.com/StarRocks/starrocks/issues/5062

## Problem Summary(Required) ：

Normalized catalog analyzer

### CODE REVIEW

**Comment by stephen-shelby** [comment]
cc @HangyuanLiu  PTAL

**Review by DorianZheng — DISMISSED** [review]
LGTM

**Review by caneGuy — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CatalogAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.starrocks.analysis.ShowStmt;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.sql.ast.AstVisitor;
+import com.starrocks.sql.ast.CatalogStmt;
+import com.starrocks.sql.ast.CreateCatalogStmt;
+import com.starrocks.sql.ast.DropCatalogStmt;
+
+public class CatalogAnalyzer {
+    public static void analyze(CatalogStmt stmt, ConnectContext session) {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CatalogAnalyzer.java:13]:
```suggestion
    public static void analyze(StatementBase stmt, ConnectContext session) {
```

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/CatalogAnalyzer.java]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java [hunk 2]
```diff
@@ -178,22 +175,9 @@ public Void visitAlterSystemStmt(AlterSystemStmt statement, ConnectContext conte
         }
 
         @Override
-        public Void visitCreateCatalogStatement(CreateCatalogStmt statement, ConnectContext context) {
-            statement.analyze();
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java:182]: CatalogAnalyzer.analyze(statement)

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java [hunk 3]
```diff
@@ -178,22 +175,9 @@ public Void visitAlterSystemStmt(AlterSystemStmt statement, ConnectContext conte
...
-            return null;
-        }
-
-        @Override
-        public Void visitDropCatalogStatement(DropCatalogStmt statement, ConnectContext context) {
-            statement.analyze();
+        public Void visitCatalogStatement(StatementBase statement, ConnectContext context) {
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java:178]:
```suggestion
        public Void visitDropCatalogStatement(DropCatalogStmt statement, ConnectContext context) {
```

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java (hunk 2 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java:182]: done

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java (hunk 3 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 28 / 40 (70.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|3|33.33%|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/CatalogAnalyzer.java|20|29|68.97%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|3|3|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/CreateCatalogStmt.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/qe/DdlExecutor.java|1|1|100.00%|

**Review by HangyuanLiu — APPROVED** [review]

**Review by caneGuy — APPROVED** [review]

**Pull request merged**